### PR TITLE
EVG-15081: Add GitHub Page

### PR DESCRIPTION
**Pull request by sophstad**
EVG-15081

### Description
- Add GitHub page
-  Still to do:
    - "Git Tag Version Definitions" section will be completed in EVG-16117 as there was design required
    - "Patch Trigger Aliases" section will be completed in EVG-15084, as the "Aliases" page is where the data is actually managed

### Screenshots
image
image
image

### Testing 
Add integration tests. To preview locally when running a dev server using the latest main branch of evergreen:

- http://localhost:3000/project/evergreen/settings/access to view a project that inherits from a repo
- http://localhost:3000/project/spruce/settings/access to view a project that does not inherit from a repo
- http://localhost:3000/project/602d70a2b2373672ee493184/settings/access to view a repository's settings

### CODE REVIEW

File: src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts
Comment (ablack12) [inline src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts:94]: I realize I wrote this text 🤦‍♀️  but it should probably say "regex/tag list". We could also tweak this to be more like the later definitions:
For patches created from GitHub pull requests, Evergreen will schedule only the tasks and variants matching the tags/regexes definitions. All regular expressions must be valid Golang regular expressions.

File: src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts
Comment (ablack12) [inline src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts:86]: Wanted to call out that this behaves slightly different than the below (disabling Pull Request Testing doesn't prevent you from modifying github aliases).
image

File: src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts
Comment (sophstad) [reply src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts:108]: Makes sense! To clarify for a project that is attached to a repo: should we only provide this warning if neither the project nor the repo has aliases defined?

File: src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts
Comment (ablack12) [reply src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts:71]: I think that message should be tweaked slightly since Git Tag Versions and Github Checks also aren't available. 
Maybe the messaging could either give the full list or say "Github Features"? (The latter would probably only be reasonable if the content was disabled and not hidden, since the user may not know what it is they're missing out on.)

Possibly we could also add a disclaimer "Please reach out to #evergreen-users if you are still experiencing issues.

File: src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts
Comment (ablack12) [reply src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts:108]: If the project chooses "Override Repo Configuration" then it doesn't matter if the repo has aliases defined so we can still show the message.

Comment (ablack12) [reply src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts:108]: Actually also: There is now a [VersionConfigEnabled boolean](https://github.com/evergreen-ci/evergreen/blob/8a425038834326c212d65289e0c9e80e48d07e7e/rest/model/project.go#L466) (Malik created https://jira.mongodb.org/browse/EVG-16185 to add this). It would be awesome if this showed as an _error_ if version control is not enabled, and a warning if it is. In which case maybe this validation should be tacked onto that ticket.

File: src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts
Comment (sophstad) [reply src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts:83]: Discussed offline, EVG-16194 and EVG-16195 have been created to address this issue

File: src/components/ProjectSettingsTabs/AliasRow.tsx
Comment (malikchaya2) [inline src/components/ProjectSettingsTabs/AliasRow.tsx:38]: Is the lone comma at the beginning intentional?

Comment (sophstad) [reply src/components/ProjectSettingsTabs/AliasRow.tsx:38]: Yes! It's [array destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#array_destructuring) in which the zeroth element is ignored.

File: src/components/SpruceForm/FieldTemplates.tsx
Comment (minnakt) [inline src/components/SpruceForm/FieldTemplates.tsx:133]: seems like this typo came back?!  🤔

File: src/components/ProjectSettingsTabs/AliasRow.tsx
Comment (minnakt) [inline src/components/ProjectSettingsTabs/AliasRow.tsx:106]: Is there any reason these `id`s can't be `classNames`? Or else there's a lot of elements with the same non-unique `id` (also applies to lines `135-137`)

File: src/components/ProjectSettingsTabs/GithubCommitQueue/GithubCommitQueueTab.tsx
Comment (minnakt) [inline src/components/ProjectSettingsTabs/GithubCommitQueue/GithubCommitQueueTab.tsx:73]: Can add return types here -- `boolean` for the first two functions... honestly not sure for the last one, I guess just leave it if it's not easy to figure out 😅

File: src/components/ProjectSettingsTabs/AliasRow.tsx
Comment (minnakt) [inline src/components/ProjectSettingsTabs/AliasRow.tsx:70]: Would it make sense to default to the `Tags` tab if the regex field is empty? (At least for this page, all of the `SegmentedControl`s default to the `Regex` tab and I was kind of confused why it was empty until I clicked on it)

https://user-images.githubusercontent.com/47064971/151233514-db5add5a-d4c0-4516-8db5-4bf312480162.mov

File: src/components/ProjectSettingsTabs/AliasRow.tsx
Comment (malikchaya2) [reply src/components/ProjectSettingsTabs/AliasRow.tsx:38]: nice!

File: src/components/ProjectSettingsTabs/AliasRow.tsx
Comment (malikchaya2) [inline src/components/ProjectSettingsTabs/AliasRow.tsx:86]: how would you feel about not clearing it between toggles? On the plugins page where there's a toggle between custom ticket creation versus default ticket creation, instead of clearing the form, I leave it there in case the user changes their mind and wants to come back to it, and instead make sure to only save the appropriate one when the form is saved.

Comment (sophstad) [reply src/components/ProjectSettingsTabs/AliasRow.tsx:86]: Definitely would prefer not clearing it—I went with this approach because it was significantly easier to implement 😅 I could try looking into it a bit more, or we could create a ticket as an improvement for later. Open to either!

Comment (malikchaya2) [reply src/components/ProjectSettingsTabs/AliasRow.tsx:86]: ah I definitely hear that! I would definitely suggest a separate ticket as opposed to looking into to it now :)

**Review by malikchaya2 — APPROVED** [review]

File: src/components/ProjectSettingsTabs/AliasRow.tsx
Comment (sophstad) [reply src/components/ProjectSettingsTabs/AliasRow.tsx:89]: EVG-16214

File: src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts
Comment (sophstad) [reply src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts:108]: EVG-16215 can handle validation for both cases.

File: cypress/integration/projectSettings/project_settings.ts
Comment (ablack12) [inline cypress/integration/projectSettings/project_settings.ts:249]: I would suggest  that these also hide when Trigger Versions With Git Tags is disabled (to be consistent with CQ and PR).

File: src/components/ProjectSettingsTabs/GithubCommitQueue/GithubCommitQueueTab.tsx
Comment (ablack12) [inline src/components/ProjectSettingsTabs/GithubCommitQueue/GithubCommitQueueTab.tsx:59]: image

FYI this doesn't prevent the user from saving the page with Enabled set for these things, and it should be, but I'm not sure if that's what line 71 is about.

**Review by ablack12 — APPROVED** [review]
Left a couple of comments but no need to block merge on another review from me!

File: src/components/ProjectSettingsTabs/GithubCommitQueue/GithubCommitQueueTab.tsx
Comment (sophstad) [reply src/components/ProjectSettingsTabs/GithubCommitQueue/GithubCommitQueueTab.tsx:59]: Yeah, the entire page will be disabled if GitHub webhooks are not enabled once we fix the EVG-16208 bug. For the meantime I'm preventing disabling so that the page can be tested.

Comment (ablack12) [reply src/components/ProjectSettingsTabs/GithubCommitQueue/GithubCommitQueueTab.tsx:59]: Oh right as we literally discussed 🤦‍♀️

**Pull request merged**
